test/e2e/install/install_test.go

```diff
@@ -58,8 +60,21 @@ var _ = Describe("[Install] continuousIntegration", func() {
 	})
 
 	Context("with a running cluster with all the components", func() {
-		It("should perform overall deployment and verify the condition is reported as available", func() {
-			deployedObj := overallDeployment()
+		var deployedObj []client.Object
+		BeforeEach(func() {
```

Comment:
why was this moved to `beforeeach`?
I realize it could be confusing (and actually wrong) to leverage the side effect the suite is leaving TAS installed on the system; this should be at least documented.

Reply:
documented -> not in this PR (separate one)

Reply:
Well I just saw newly tests introduced in the package were following this pattern and I though it would be a good idea to make them all work in the same way and also clean-up the system after the test. 
I did not add any comment because I though it would not be needed as I was just using an already used pattern.
I will add some comments.

Reply:
ok, no problem. In this case we need a separate PR to document we intentionally leave the cluster state mutated after the install suite runs, and to explain why we do like we do.

Reply:
Ok I though you want to comment why I was changing this to BeforeEach ... Now I see the point.